Sources_v3/StreamChatUI/ChatChannel/MessageComposer/MessageComposerSuggestionsViewController.swift

```diff
@@ -134,47 +123,111 @@ open class MessageComposerSuggestionsViewController<ExtraData: ExtraDataTypes>:
+            forCellWithReuseIdentifier: uiConfig.messageComposer.suggestionsCommandCollectionViewCell.reuseId
+        )
+    }
+
+    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        commands.count
+    }
+
+    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        let cell = collectionView.dequeueReusableCell(
+            withReuseIdentifier: MessageComposerCommandCollectionViewCell<ExtraData>.reuseId,
+            for: indexPath
+        ) as! MessageComposerCommandCollectionViewCell<ExtraData>
+
+        cell.uiConfig = uiConfig
+        cell.commandView.command = commands[indexPath.row]
+
+        return cell
+    }
+}
+
+class SuggestionsMentionDataSource<ExtraData: ExtraDataTypes>: NSObject,
+    UICollectionViewDataSource,
+    ChatUserSearchControllerDelegate {
+    var collectionView: MessageComposerSuggestionsCollectionView<ExtraData>
+    var searchController: _ChatUserSearchController<ExtraData>
+
+    var uiConfig: UIConfig<ExtraData> {
+        collectionView.uiConfig()
+    }
+
+    init(
+        collectionView: MessageComposerSuggestionsCollectionView<ExtraData>,
+        searchController: _ChatUserSearchController<ExtraData>
+    ) {
+        self.collectionView = collectionView
+        self.searchController = searchController
+        super.init()
+        registerCollectionViewCell()
+    }
+
+    private func registerCollectionViewCell() {
+        collectionView.register(
+            uiConfig.messageComposer.suggestionsMentionCollectionViewCell,
+            forCellWithReuseIdentifier: uiConfig.messageComposer.suggestionsMentionCollectionViewCell.reuseId
+        )
+    }
+
+    // ✅ Init with collectionView (register cells)
+    // -> didChangeUsers -> Apply changes to collectionView with batchupdates
+
+    // how to add query to userSearchVC ->
+    // call userSearchController.search(term: nil)
+    //
+
+    // MARK: - CollectionViewDataSource
+
+    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        searchController.users.count
+    }
+
+    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        let cell = collectionView.dequeueReusableCell(
+            withReuseIdentifier: MessageComposerMentionCollectionViewCell<ExtraData>.reuseId,
+            for: indexPath
+        ) as! MessageComposerMentionCollectionViewCell<ExtraData>
+
+        let user = searchController.users[indexPath.row]
+        cell.mentionView.content = (user.name!, user.id, user.imageURL, true)
+        cell.uiConfig = uiConfig
+        return cell
+    }
+
+    // MARK: - ChatUserSearchControllerDelegate
+
+    func controller(
+        _ controller: ChatUserSearchController,
+        didChangeUsers changes: [ListChange<ChatUser>]
+    ) {
+        collectionView.reloadData()
```

Comment:
You can also animate the changes here. See the boilerplate elsewhere in the SDK

Reply:
When I do `performBatchUpdates` with some animation the changes, the observer doesn't talk too much with the animation being smooth. I can see what I can do, but I'd say for now this looks better than fade-out animation, because the cell disappears and the frame immediately changes :) Thanks